vortex-expr/src/binary.rs

```diff
@@ -82,6 +83,66 @@ pub(crate) mod proto {
     }
 }
 
+impl AnalysisExpr for BinaryExpr {
+    fn prune_expr(&self, catalog: &mut dyn StatsCatalog) -> Option<ExprRef> {
+        match self.operator {
+            Operator::Eq => {
+                let min_lhs = self.lhs.min(catalog);
+                let max_lhs = self.lhs.max(catalog);
+
+                let min_rhs = self.rhs.min(catalog);
+                let max_rhs = self.rhs.max(catalog);
+
+                let left = min_lhs.zip_with(max_rhs, gt);
+                let right = min_rhs.zip_with(max_lhs, gt);
+                left.into_iter().chain(right).reduce(or)
+            }
+            Operator::NotEq => {
+                let min_lhs = self.lhs.min(catalog);
+                let max_lhs = self.lhs.max(catalog);
+
+                let min_rhs = self.rhs.min(catalog);
+                let max_rhs = self.rhs.max(catalog);
+
+                let left = min_lhs.zip_with(max_rhs, eq);
+                let right = max_lhs.zip_with(min_rhs, eq);
+                left.into_iter().chain(right).reduce(and)
```

Comment:
Now that I see what the reduce is doing, I think this is wrong. Just because the right-hand-side doesn't have a min or max doesn't mean that we can use `left` alone. Specifically, `min(lhs) == max(rhs)`, alone, is not sufficient to conclude that `lhs != rhs`. For example: $lhs \in [1, 2], rhs \in [0, 1]$

Reply:
I think, except for the `or` cases, we should use question mark notation both for the correct semantics (we need both sides of the and) and for legibility.

Reply:
correct, this is wrongg